lib/core/modules/coderunner/codeRunner.js

```diff
@@ -1,76 +1,135 @@
   constructor(options) {
+    this.config = options.config;
     this.plugins = options.plugins;
     this.logger = options.logger;
     this.events = options.events;
     this.ipc = options.ipc;
     this.commands = [];
     this.runCode = new RunCode();
-    let self = this;
+    this.registerIpcEvents();
+    this.IpcClientListen();
+    this.registerEvents();
+    this.registerCommands();
+    this.registerEmbarkJs();
+  }
 
-    if (this.ipc.isServer()) {
-      this.ipc.on('runcode:getCommands', (_err, callback) => {
-        let result = {web3Config: self.runCode.getWeb3Config(), commands: self.commands};
-        callback(null, result);
-      });
+  registerIpcEvents() {
+    if (!this.ipc.isServer()) {
+      return;
     }
 
-    if (this.ipc.isClient() && this.ipc.connected) {
-      this.ipc.listenTo('runcode:newCommand', function (command) {
-        if (command.varName) {
-          self.events.emit("runcode:register", command.varName, command.code);
-        } else {
-          self.events.request("runcode:eval", command.code);
-        }
-      });
+    this.ipc.on('runcode:getCommands', (_err, callback) => {
+      let result = {web3Config: this.runCode.getWeb3Config(), commands: this.commands};
+      callback(null, result);
+    });
+  }
+
+  IpcClientListen() {
+    if (!this.ipc.isClient() || !this.ipc.connected) {
+      return;
     }
 
-    this.events.on("runcode:register", (varName, code) => {
-      if (self.ipc.isServer() && varName !== 'web3') {
-        self.commands.push({varName, code});
-        self.ipc.broadcast("runcode:newCommand", {varName, code});
+    this.ipc.listenTo('runcode:newCommand', (command) => {
+      if (command.varName) {
+        this.events.emit("runcode:register", command.varName, command.code);
+      } else {
+        this.events.request("runcode:eval", command.code);
       }
-      self.runCode.registerVar(varName, code);
     });
+  }
 
+  registerEvents() {
+    this.events.on("runcode:register", this.registerVar.bind(this));
+  }
+
+  registerCommands() {
     this.events.setCommandHandler('runcode:getContext', (cb) => {
-      cb(self.runCode.context);
+      cb(this.runCode.context);
     });
+    this.events.setCommandHandler('runcode:eval', this.evalCode.bind(this));
+  }
 
-    this.events.setCommandHandler('runcode:eval', (code, cb, forConsoleOnly = false) => {
-      if (!cb) {
-        cb = function() {};
-      }
-      const awaitIdx = code.indexOf('await');
-      if (awaitIdx > -1) {
-        if (awaitIdx < 2) {
-          let end = code.length;
-          if (code[end - 1] === ';') {
-            end--; // Remove the `;` because we add function calls
-          }
-          code = code.substring(5, end); // remove await keyword
-        } else {
-          code = `(async function() {${code}})();`;
+  registerEmbarkJs() {
+    this.registerVar('IpfsApi', IpfsApi, false);
+    this.registerVar('Web3', Web3, false);
+    this.registerVar('EmbarkJS', EmbarkJS, false);
```

Comment:
can we make this through an API? for example see https://github.com/embark-framework/embark/blob/develop/lib/modules/ipfs/index.js#L106

This way codeRunner remains 'agnostic'

Reply:
agree, I am moving it to console